### ISSUE TITLE
some stream objects violate the IByteStream interface

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3163]
|<img alt="camrdale's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| camrdale reported|
|-|-|
|Trac ID|trac#3163|
|Type|defect|
|Created|2008-04-04 20:31:46Z|

IByteStream specifies that the length attribute is "how much data is in this stream".

If you take this to mean how much is in the stream to begin with (which seems the most sensible interpretation), then some objects (e.g. FileStream) violate the interface as they modify the length attribute as data is read to indicate how much data is left in the stream.

If you take this to mean how much is LEFT in the stream, then some objects (e.g. ProducerStream) violate the interface as they only set the length at the beginning and don't update it as data is read.

<details><summary>Searchable metadata</summary>

```
trac-id__3163 3163
type__defect defect
reporter__camrdale camrdale
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1207341106000000 1207341106000000
changetime__1297616987000000 1297616987000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

Not going to fix this in web2 (See [#4821](https://github.com/twisted/twisted/issues/4821)).  Nothing corresponding to streams in Twisted Web, so not retargeting this ticket at the web component.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
